src/main/java/org/aksw/simba/squirrel/analyzer/impl/TurtleAutomata.java

```diff
@@ -0,0 +1,76 @@
+package org.aksw.simba.squirrel.analyzer.impl;
+
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFLanguages;
+
+/**
+ * Default implementation of a finite state machine.
+ * This class is immutable and thread-safe.
+ */
+
+public final class TurtleAutomata implements FiniteStateMachine {
+
+    /**
+     * Current state.
+     */
+    private State current;
```

Comment:
The `TurtleAutomata` and the `RdfAutomata` are sharing attributes and methods which are exactly the same. You could introduce an `AbstractAutomata` class which implements them. That would make your code cleaner and a further extension (e.g., for JSON-LD) easier. 

It would be even possible to create a general `Automata` class since they are all working in the same way. Only the transitions and the mime type are different and both is set in the factory.